dirty_cat/_deduplicate.py

```diff
@@ -0,0 +1,175 @@
+"""
+Implements deduplication based on clustering string distance matrices.
+This works best if there is a number of underlying categories that
+sometimes appear in the data with small variations and/or misspellings.
+"""
+from typing import List, Literal, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from scipy.cluster.hierarchy import fcluster, linkage
+from scipy.spatial.distance import pdist, squareform
+from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
+from sklearn.metrics import silhouette_score
+
+
+def compute_ngram_distance(
```

Comment:
I think the three support functions guess_clusters, compute_ngram_distance, create_spelling_correction should be made private (add a _ before the name)